### ISSUE TITLE
Modcontent crashes with a `NullReferenceException` 

### ISSUE THREAD

**Issue by dragunoff** [OpenRA/OpenRA#17565]
### Issue Summary
Modcontent crashes with a `NullReferenceException` when trying to access it after clicking around in game menus.

#### System Information
- **Operating System:** Ubuntu 19.10
- **.NET / Mono Version:** [e.g. .NET 4.7.1, Mono 4.6.2, ...]
- **OpenRA Version:** release-20191117, bleed
- **Mod:** Any

#### Additional Information:
- Steps to reproduce
1. Click "SIngleplayer", 
1. Click "Skirmish", click "Back"
1. Click "Missions", click "Back", click "Back" (to main menu)
1. Click "Multiplayer", click "Back"  (to main menu)
1. Click "Settings", click "Back"  (to main menu)
1. Click "Extras"
1. Click "Replays", click "Back", click "Back"  (to main menu)
1. Click "Manage content"

- Logs
<details>

```
Loading mod: modcontent
Exception of type `System.NullReferenceException`: Object reference not set to an instance of an object
  at OpenRA.Mods.Common.Traits.MapOptionsInfo.OpenRA.Traits.IRulesetLoaded<OpenRA.ActorInfo>.RulesetLoaded (OpenRA.Ruleset rules, OpenRA.ActorInfo info) [0x00000] in <123db89d1f414b4686fb3ebc9896b24e>:0 
  at OpenRA.Ruleset..ctor (OpenRA.IReadOnlyDictionary`2[TKey,TValue] actors, OpenRA.IReadOnlyDictionary`2[TKey,TValue] weapons, OpenRA.IReadOnlyDictionary`2[TKey,TValue] voices, OpenRA.IReadOnlyDictionary`2[TKey,TValue] notifications, OpenRA.IReadOnlyDictionary`2[TKey,TValue] music, OpenRA.TileSet tileSet, OpenRA.Graphics.SequenceProvider sequences, OpenRA.IReadOnlyDictionary`2[TKey,TValue] modelSequences) [0x00072] in <742652ed86ed424f90d92f1e5a818a0c>:0 
  at OpenRA.Ruleset.LoadDefaultsForTileSet (OpenRA.ModData modData, System.String tileSet) [0x00021] in <742652ed86ed424f90d92f1e5a818a0c>:0 
  at OpenRA.MapPreview+InnerData+<>c__DisplayClass28_0.<SetRulesetGenerator>b__0 () [0x0007f] in <742652ed86ed424f90d92f1e5a818a0c>:0 
  at System.Lazy`1[T].ViaFactory (System.Threading.LazyThreadSafetyMode mode) [0x00043] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 
  at System.Lazy`1[T].ExecutionAndPublication (System.LazyHelper executionAndPublication, System.Boolean useDefaultConstructor) [0x00022] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 
  at System.Lazy`1[T].CreateValue () [0x00074] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 
  at System.Lazy`1[T].get_Value () [0x0000a] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 
  at OpenRA.MapPreview+InnerData.get_Rules () [0x0000a] in <742652ed86ed424f90d92f1e5a818a0c>:0 
  at OpenRA.MapPreview.get_Rules () [0x00000] in <742652ed86ed424f90d92f1e5a818a0c>:0 
  at OpenRA.MapPreview.PreloadRules () [0x00000] in <742652ed86ed424f90d92f1e5a818a0c>:0 
  at OpenRA.Mods.Common.Widgets.Logic.MissionBrowserLogic+<>c__DisplayClass21_1.<.ctor>b__10 () [0x0001d] in <123db89d1f414b4686fb3ebc9896b24e>:0 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00014] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 
  at System.Threading.ThreadHelper.ThreadStart () [0x00008] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 

[ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object
  at OpenRA.Mods.Common.Traits.MapOptionsInfo.OpenRA.Traits.IRulesetLoaded<OpenRA.ActorInfo>.RulesetLoaded (OpenRA.Ruleset rules, OpenRA.ActorInfo info) [0x00000] in <123db89d1f414b4686fb3ebc9896b24e>:0 
  at OpenRA.Ruleset..ctor (OpenRA.IReadOnlyDictionary`2[TKey,TValue] actors, OpenRA.IReadOnlyDictionary`2[TKey,TValue] weapons, OpenRA.IReadOnlyDictionary`2[TKey,TValue] voices, OpenRA.IReadOnlyDictionary`2[TKey,TValue] notifications, OpenRA.IReadOnlyDictionary`2[TKey,TValue] music, OpenRA.TileSet tileSet, OpenRA.Graphics.SequenceProvider sequences, OpenRA.IReadOnlyDictionary`2[TKey,TValue] modelSequences) [0x00072] in <742652ed86ed424f90d92f1e5a818a0c>:0 
  at OpenRA.Ruleset.LoadDefaultsForTileSet (OpenRA.ModData modData, System.String tileSet) [0x00021] in <742652ed86ed424f90d92f1e5a818a0c>:0 
  at OpenRA.MapPreview+InnerData+<>c__DisplayClass28_0.<SetRulesetGenerator>b__0 () [0x0007f] in <742652ed86ed424f90d92f1e5a818a0c>:0 
  at System.Lazy`1[T].ViaFactory (System.Threading.LazyThreadSafetyMode mode) [0x00043] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 
  at System.Lazy`1[T].ExecutionAndPublication (System.LazyHelper executionAndPublication, System.Boolean useDefaultConstructor) [0x00022] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 
  at System.Lazy`1[T].CreateValue () [0x00074] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 
  at System.Lazy`1[T].get_Value () [0x0000a] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 
  at OpenRA.MapPreview+InnerData.get_Rules () [0x0000a] in <742652ed86ed424f90d92f1e5a818a0c>:0 
  at OpenRA.MapPreview.get_Rules () [0x00000] in <742652ed86ed424f90d92f1e5a818a0c>:0 
  at OpenRA.MapPreview.PreloadRules () [0x00000] in <742652ed86ed424f90d92f1e5a818a0c>:0 
  at OpenRA.Mods.Common.Widgets.Logic.MissionBrowserLogic+<>c__DisplayClass21_1.<.ctor>b__10 () [0x0001d] in <123db89d1f414b4686fb3ebc9896b24e>:0 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00014] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 
  at System.Threading.ThreadHelper.ThreadStart () [0x00008] in <d2ec5c92492f4d6ba8c422bdf574b786>:0 
AL lib: (EE) alc_cleanup: 1 device not closed
```

</details>

**Comment by pchote** [comment]
Probably a duplicate of #17314 and #17369, which are probably timing / threading related, and therefore difficult to reproduce without artificially slowing things down.

**Comment by PunkPun** [comment]
Is it stll an issue?

**Comment by dragunoff** [comment]
I don't know. It happened only once back then. I thought the crash logs would reveal what might have gone wrong. I'd say it's fine to close this issue if nothing similar has been reported since.

**Comment by PunkPun** [comment]
Alright
